File: validate-ip-address..java
public static String validIPAddress(String IP) {
        String[] ipv4 = IP.split("\\.",-1);
        String[] ipv6 = IP.split("\\:",-1);
        if(IP.chars().filter(ch -> ch == '.').count() == 3){
            for(String s : ipv4) if(isIPv4(s)) continue;else return "Neither"; return "IPv4";
        }
        if(IP.chars().filter(ch -> ch == ':').count() == 7){
            for(String s : ipv6) if(isIPv6(s)) continue;else return "Neither";return "IPv6";
        }
        return "Neither";
    }
    public static boolean isIPv4 (String s){
         try{ return String.valueOf(Integer.valueOf(s)).equals(s) && Integer.parseInt(s) >= 0 && Integer.parseInt(s) <= 255;}
         catch (NumberFormatException e){return false;}
    }
    public static boolean isIPv6 (String s){
        if (s.length() > 4) return false;
        try {return Integer.parseInt(s, 16) >= 0  && s.charAt(0) != '-';}
        catch (NumberFormatException e){return false;}
    }

File: validate-ip-address.py
class Solution:
    def validIPAddress(self, queryIP: str) -> str:
        colon = queryIP.count(":")
        dot = queryIP.count(".")
        if (dot and colon) or (not colon and not dot):
            return "Neither"
        elif dot and "00" == queryIP[:2] or ".00" in queryIP:
            return "Neither"
        elif "::" in queryIP or ".." in queryIP:
            return "Neither"
        elif dot != 3 and colon != 7  or queryIP[-1] == "." or queryIP[-1] == ":":
            return "Neither"
        elif queryIP[0] == "." or queryIP[0] == ":":
            return "Neither"
        
        if dot == 3 and len(queryIP) < 20:
            left = 0
            index = []
            if queryIP[0] == "0" and queryIP[1] != ".":
                return "Neither"
            for right in range(1,len(queryIP)-1):
                if queryIP[right] == ".":
                    index.append(right)
                    if (right-left-1) > 4: 
                        return "Neither"
                    left = right
                
                elif queryIP[right] == "0" and right-1 == left:
                    if queryIP[right+1] != '0' and queryIP[right+1] != '.':
                        return "Neither"
                elif not queryIP[right].isdigit():
                    return "Neither"
            if int(queryIP[:index[0]]) > 255 or int(queryIP[index[0]+1:index[1]]) > 255 or int(queryIP[index[1]+1:index[2]]) > 255 or int(queryIP[index[2]+1:]) > 255:
                return "Neither"
            
                
            return "IPv4"
        # return 
        # print(queryIP[0])
        # print()
        elif colon == 7 and len(queryIP) < 40:
            left = 0
            # index = []
            # print(left)

            for right in range(1,len(queryIP)-1):

                if queryIP[right] == ":":
                    # index.append(right)
                    if (right-left-1) > 4: 
                        return "Neither"
                    left = right
                elif not queryIP[right].isdigit():
                    if queryIP[right].lower() > 'f' or queryIP[0].lower() > 'f':
                        return "Neither"

            # print(len(queryIP),left)
            if len(queryIP) - left - 1 > 4:
                return "Neither"
            return "IPv6"
        return "Neither"